configdialog.cpp

```diff
@@ -1530,3 +1557,128 @@ LRESULT CALLBACK AddPredefinedPaletteDialog(HWND dialog, UINT message, WPARAM wp
+				ListBox_AddString(disabled_patterns_listbox, pattern_strings.at(pattern).c_str());
+			}
+
+			HWND enable_button = GetDlgItem(dialog, IDC_ENABLE_PATTERN);
+			HWND disable_button = GetDlgItem(dialog, IDC_DISABLE_PATTERN);
+
+			EnableWindow(enable_button, false);
+			EnableWindow(disable_button, false);
+
+			if (enabled_patterns.size() <= 0) {
+				HWND no_patterns_enabled_text = GetDlgItem(dialog, IDC_NO_PATTERNS_ENABLED_TEXT);
+				HWND ok_button = GetDlgItem(dialog, IDOK);
+				ShowWindow(no_patterns_enabled_text, SW_NORMAL);
+				EnableWindow(ok_button, FALSE);
+			}
+
+			return TRUE;
+		}
+		case WM_COMMAND: {
+			HWND enabled_patterns_listbox = GetDlgItem(dialog, IDC_PATTERN_SELECTOR_ENABLED);
+			HWND disabled_patterns_listbox = GetDlgItem(dialog, IDC_PATTERN_SELECTOR_DISABLED);
+			HWND enable_button = GetDlgItem(dialog, IDC_ENABLE_PATTERN);
+			HWND disable_button = GetDlgItem(dialog, IDC_DISABLE_PATTERN);
+			HWND no_patterns_enabled_text = GetDlgItem(dialog, IDC_NO_PATTERNS_ENABLED_TEXT);
+			HWND ok_button = GetDlgItem(dialog, IDOK);
+
+			switch (LOWORD(wparam)) {
+				case IDOK: {
+					PatternRepository::save_disabled_patterns(disabled_patterns);
+					[[fallthrough]];
+				}
+				case IDCANCEL: {
+					EndDialog(dialog, LOWORD(wparam) == IDOK);
+					return TRUE;
+				}
+				case IDC_PATTERN_SELECTOR_ENABLED: {
+					switch (HIWORD(wparam)) {
+						case LBN_SELCHANGE: {
+							ListBox_SetSel(disabled_patterns_listbox, FALSE, -1);
+							EnableWindow(enable_button, FALSE);
+							EnableWindow(disable_button, ListBox_GetSelCount(enabled_patterns_listbox) != 0);
+
+							break;
+						}
+					}
+					return FALSE;
+				}
+				case IDC_PATTERN_SELECTOR_DISABLED: {
+					switch (HIWORD(wparam)) {
+						case LBN_SELCHANGE: {
+							ListBox_SetSel(enabled_patterns_listbox, FALSE, -1);
+							EnableWindow(enable_button, ListBox_GetSelCount(disabled_patterns_listbox) != 0);
+							EnableWindow(disable_button, FALSE);
+
+							break;
+						}
+					}
+					return FALSE;
+				}
+				case IDC_ENABLE_PATTERN: {
+					while (ListBox_GetSelCount(disabled_patterns_listbox) > 0) {
+						int selection = 0;
+						ListBox_GetSelItems(disabled_patterns_listbox, 1, &selection);
+						auto &selected_pattern = disabled_patterns.at(selection);
+						enabled_patterns.push_back(selected_pattern);
+						ListBox_AddString(enabled_patterns_listbox, pattern_strings.at(selected_pattern).c_str());
+						ListBox_DeleteString(disabled_patterns_listbox, selection);
+						disabled_patterns.erase(std::find(disabled_patterns.begin(), disabled_patterns.end(), selected_pattern));
+					}
+					EnableWindow(enable_button, FALSE);
+
+					if (enabled_patterns.size() > 0) {
+						ShowWindow(no_patterns_enabled_text, SW_HIDE);
+						EnableWindow(ok_button, TRUE);
+					}
+
+					return FALSE;
+				}
+				case IDC_DISABLE_PATTERN: {
+					while (ListBox_GetSelCount(enabled_patterns_listbox) > 0) {
+						int selection = 0;
```

Comment:
Add some line breaks in there man, it makes a huge difference for readability. Here's how I would space things out (trying to keep the same "steps" of the operation visually together)

```cpp
int selection = 0;
ListBox_GetSelItems(disabled_patterns_listbox, 1, &selection);

auto &selected_pattern = disabled_patterns.at(selection);
enabled_patterns.push_back(selected_pattern);
ListBox_AddString(enabled_patterns_listbox, pattern_strings.at(selected_pattern).c_str());

ListBox_DeleteString(disabled_patterns_listbox, selection);
disabled_patterns.erase(std::find(disabled_patterns.begin(), disabled_patterns.end(), selected_pattern));
```